docs/kind-php.md

```diff
@@ -0,0 +1,437 @@
+server along with some workers or webserver, all from a single `Dockerfile`.
+
+zbuild helps you there:
+
+1. it supports multi-stage workflows, so you can easily create a base config
+and derive it to make specialized stages. When the build process kicks off, 
+the final build config is resolved by merging each stage with its parent, until
+the `base` stage ;
+
+2. you can also mark stages as dev. This will build lighter images that include
+only what's needed to start a container with bind-mounts (no composer install,
+no autoload dump, no post install steps, etc...) ;
+
+As an example, here're two visual representations of the build steps executed
+by zbuild. The first example has been generated from a `dev` stage:
+
+![](diagrams/php-stage-dev.png)
+
+An this one has been generated from a non-dev stage (these come from zbuild 
+test suite):
+
+![](diagrams/php-stage-prod.png)
+
+## Config inference
+
+When inference is enabled, zbuild automatically:
+
+* Add `apcu` and `opcache` extensions for non-dev stages, when FPM mode is
+enabled ;
+* Add extensions requirements from your `composer.json` and `composer.lock`
+files ;
+* Add `zip` extension (as it's required by composer) ;
+* Add `libpcre3-dev`, `unzip` and `git` system packages ;
+* Add known system packages requirements for the native extensions to install ;
+
+## Build process
+
+The images build from php kinds have following steps:
+
+* Install system packages ;
+* Install PHP extensions ;
+* Copy external files ;
+* Create `/app` and `/composer` paths ;
+* At this point, the build process switch to user `1000` ;
+* Copy config files ;
+* Install global dependencies ;
+
+Moreover, when the target stage is non-dev, following steps also happen:
+
+* Install project dependencies ;
+* Copy sources files ;
+* Execute post install steps ;
+
+Also note that uid 1000 is automatically set as the default user.
+
+## Locking
+
+When using `zbuild update` to create or update your lockfile, the base image
+digest is resolved and for each stage, extensions and system packages are
+pinned to a specific version.
+
+For the base image, zbuild uses either `version` and `fpm` parameters to
+determine what's the base image or directly the `base` parameter (see below).
+In both cases, the digest of the image is resolved. As such, even if the image
+tag used changes, zbuild will continue to use the locked version.
+
+For the extensions, zbuild don't lock native extensions as they're installed
+using `docker-php-ext-install` which is actually using the PHP source code 
+bundled inside the base image to compile these extensions. However, for community
+extensions, [notpecl](https://github.com/NiR-/notpecl) is used to resolve the
+version constraint and the last version available matching that version
+constraint is locked.
+
+## Assets and webserver
+
+Sometimes you need to run some PHP commands to build some static assets. To
+help you do so, zbuilfiles for php kind lets you embed a webserver definition:
```

Comment:
I'm not sure that this second sentence is related to the previous one (i.e. `to help you to do so` : the webserver doesn't help to run PHP commands IMHO, neither to build static assets).
It helps to serve static assets tho.